src/libcore/num/mod.rs

```diff
@@ -2037,6 +2038,120 @@ assert_eq!((-a).rem_euclid(-b), 1);
             }
         }
 
+        doc_comment! {
+            concat!("Returns the logarithm of the number with respect to an arbitrary base.
+
+Returns `None` if the number is smaller than 1.
+
+The result may not be optimized owing to implementation details;
+`self.log2()` can produce results more efficiently for base 2, and
+`self.log10()` can produce results more efficiently for base 10.
+
+# Examples
+
+```
+use std::", stringify!($SelfT), ";
+
+let five = 5", stringify!($SelfT), ";
+
+// log5(5) == 1
+let result = five.log(5);
+
+assert_eq!(result, Some(1));
+```"),
+            #[unstable(feature = "int_log", issue = "none")]
+            #[must_use = "this returns the result of the operation, \
+                          without modifying the original"]
+            #[inline]
+            pub fn log(self, base: Self) -> Option<Self> {
+                // based on https://people.csail.mit.edu/jaffer/III/ilog.pdf
+                fn ilog<T>(n: &mut T, m: T, b: T, k: T) -> T
+                    where T: PartialOrd + Div<Output=T> + Add<Output=T> + Mul<Output=T> + Copy
```

Comment:
Does `ilog` need to be generic? I think `fn ilog(n: &mut $SelfT, ...` should work.

Reply:
Oh nice, I didn't realize `$SelfT` would work inside the function body. Updated!